gh-103066: Add links and `help` in site.py constants (#103777)
python/cpython: Doc/library/constants.rst

   :exc:`SyntaxError`), so they can be considered "true" constants.

> .. _site-consts:

Reviewer: The objects above are all non-callable singleton or doubleton instances of their class.  (__debug__ is one of False or True).  All of the below are callable singleton or doubleton instances of custom classes in _sitebuiltins.  I would prefer that 'constants' be replaced by 'objects'/'callables'/'functions'/???, but this is secondary to the issue.

Author: I agree with your observation, they are not exactly constants. As the title of this section is "Built-in Constants", I won't make any changes, I feel it's a bit out of scope.

Reviewer 2: I would take the opportunity to improve this to say that _the site module adds several globals to the built-in namespace_

   (i.e. EOF) to exit", and when called, raise :exc:`SystemExit` with the
   specified exit code.

.. data:: help
   :noindex:

   Object that when printed, prints the message "Type help() for interactive

>    help, or help(object) for help about object.", and when called, invokes the

Reviewer: The first clause, giving the special effect of `help`, is not part of the 'help' entry in Built-in Functions. For this reason, and completion otherwise, I think this entry should be added.

>    help, or help(object) for help about object.", and when called, invokes the
>    built-in :func:`help` function.

Reviewer: The part beginning with 'invokes' is at least misleading when read as implying that there is or need-be a separate help function.
```suggestion
   help, or help(object) for help about object.", and when called,
   acts as described :func:`elsewhere <help>`.
```
